client/tests/integration/smartcontracts/multisig/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+//! Trigger to control multisignature account
+
+#![no_std]
+
+extern crate alloc;
+#[cfg(not(test))]
+extern crate panic_halt;
+
+use alloc::{collections::btree_set::BTreeSet, format};
+
+use iroha_trigger::{debug::dbg_panic, prelude::*, smart_contract::data_model::prelude::*};
+use lol_alloc::{FreeListAllocator, LockedAllocator};
+use serde::{Deserialize, Serialize};
+
+#[global_allocator]
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
+
+getrandom::register_custom_getrandom!(iroha_trigger::stub_getrandom);
+
+#[derive(Serialize, Deserialize)]
+enum Args {
+    /// Accept instruction and vote
+    Instruction(InstructionBox),
+    /// Accept vote for certain instruction
+    Vote(HashOf<InstructionBox>),
+}
+
+#[iroha_trigger::main]
+fn main(id: TriggerId, _owner: AccountId, event: EventBox) {
+    let (args, signatory): (Args, AccountId) = match event {
+        EventBox::ExecuteTrigger(event) => (
+            event
+                .args()
+                .dbg_expect("trigger expect args")
+                .try_into_any()
+                .dbg_expect("failed to parse arguments"),
+            event.authority().clone(),
+        ),
+        _ => dbg_panic("only work as by call trigger"),
+    };
+
+    let isi_hash = match &args {
+        Args::Instruction(isi) => HashOf::new(isi),
+        Args::Vote(isi_hash) => *isi_hash,
+    };
+    let votes_metadata_key: Name = format!("{isi_hash}/votes").parse().unwrap();
+    let isi_metadata_key: Name = format!("{isi_hash}/isi").parse().unwrap();
```

Comment:
That's right, as for now it's possible to discard instruction by perfroming 2 rounds of voting to remove associated isi (if we would extend voting from `isi` to `Vec<isi>` this would be single round of voting).

However we might consider might sophisticated schemes like you suggested as well.

Extend trigger arguments to accept negative votes or have associated trigger which would track metadata of multisig trigger and discard isi after some amount of blocks.

Do you think it's functionality we should implement in this PR?

Reply:
Could you explain how cancellations are currently supposed to be done?

I meant it would be nice if there was an equivalent to transaction ttl, but I don't think this PR should necessarily address this

Reply:
> Could you explain how cancellations are currently supposed to be done?

It certainly a hack, but users can submit 2 ISIs with `[remove hash/vote metadata, remove hash/votes]` for any isi which would result in effectively removing this transaction.

Reply:
Oh, they vote for `RemoveKeyValue<Trigger>` instruction. That would work